integration/cmd/historicalspeed-timezone-builder/timezone_builder.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"time"
+
+	"github.com/Telenav/osrm-backend/integration/unidbpatch"
+	"github.com/golang/glog"
+	"github.com/qedus/osmpbf"
+)
+
+type wayTimezoneInfo struct {
+	wayID          int64
+	timezone       string
+	daylightSaving string
+}
+
+func newTimezoneBuilderForUniDB(in <-chan *osmpbf.Way, out chan<- *wayTimezoneInfo) {
```

Comment:
I think here you could use the name `newTimezoneBuilder`
- Understand you want to clarify such information: Create timezone builder **when** input is UniDB
- `timezone` is not UniDB specific format, the output PBF should follow `OSM`'s standard, so this logic should for both situation.  One thing I am not sure is whether we strictly follow OSM's schema when define UniDB's rules.

Reply:
Timezone in unidb and osm are totally different representation. The former uses format like `-070`, but the latter uses [IANA time zone database time zone](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)(e.g., https://taginfo.openstreetmap.org/keys/timezone#values). So what I thought is we may need another `timezonebuilder` for OSM when necessary. Currently almost no timezone in OSM data(https://taginfo.openstreetmap.org/keys/timezone#overview), so we don't need to implement it now. 

Agree that `forUniDB` may not good enough. I'll remove it, use comments instead for the information.

Reply:
Tag difference will be an issue for our pipeline.  Ideally, we should have deep understanding about both data's spec(professional data and osm data), and when genrating data in `PBF` format, we should make the result really follow the OSM standard.  I think this could also be a great value to OSM community.

Reply:
Absolutely! Can we possible to do this?